insights/client/connection.py

```diff
@@ -175,6 +201,57 @@ def _init_session(self):
                 connection.proxy_headers = auth_map
         return session
 
+    def _http_request(self, url, method, **kwargs):
+        '''
+        Perform an HTTP request, net logging, and error handling
+
+        Parameters
+            url     - URL to perform the request against
+            method  - HTTP method, used for logging
+            kwargs  - Rest of the args to pass to the request function
+        Returns
+            HTTP response object on completion
+        Side effects
+            Calls handle_fail_rcs to handle messaging and exceptions for failure codes
+            Raises RuntimeError on timeout or failure to connect
+        '''
+        try:
+            logger.log(NETWORK, "%s %s", method, url)
+            res = self.session.request(url=url, method=method, timeout=self.config.http_timeout, **kwargs)
+            logger.log(NETWORK, "HTTP Status Code: %d", res.status_code)
+            logger.log(NETWORK, "HTTP Status Text: %s", res.reason)
+            logger.log(NETWORK, "HTTP Response Text: %s", res.text)
+            res.raise_for_status()
+            return res
+        except requests.Timeout as e:
+            logger.error(e)
+            raise TimeoutException("Connection timed out.")
+        except requests.ConnectionError as e:
+            logger.error(e)
+            raise RuntimeError("The Insights API could not be reached.")
+        except requests.exceptions.HTTPError as e:
+            logger.error(e)
+            self.handle_fail_rcs(res)
+            # python requests are falsy for failure codes, so this will work
+            #   for both true/false checks and cases when attributes need
+            #   to be accessed
+            return res
```

Comment:
Wouldn’t _handle_fail_rcs_ raise an error itself? I mean, it essentially does similar work like _raise_for_status_. So we may omit the raise-except here and just use our custom function. 🤷🏻‍♂️